File: membrane-field-core/_archive/test/logServer.js
// Simple log server - Run with: node logServer.js
// Then open testBundle.html in browser
// Logs will be written to debug.log

const http = require('http');
const fs = require('fs');
const path = require('path');

const LOG_FILE = path.join(__dirname, 'debug.log');
const PORT = 9999;

// Clear log file on startup
fs.writeFileSync(LOG_FILE, `=== Debug Log Started: ${new Date().toISOString()} ===\n\n`);

const server = http.createServer((req, res) => {
    // CORS headers
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    if (req.method === 'OPTIONS') {
        res.writeHead(200);
        res.end();
        return;
    }

    if (req.method === 'POST' && req.url === '/log') {
        let body = '';
        req.on('data', chunk => body += chunk);
        req.on('end', () => {
            try {
                const data = JSON.parse(body);
                const timestamp = new Date().toISOString().substr(11, 12);
                const line = `[${timestamp}] ${data.msg}\n`;
                fs.appendFileSync(LOG_FILE, line);
                res.writeHead(200);
                res.end('ok');
            } catch (e) {
                res.writeHead(400);
                res.end('bad request');
            }
        });
        return;
    }

    if (req.method === 'POST' && req.url === '/clear') {
        fs.writeFileSync(LOG_FILE, `=== Log Cleared: ${new Date().toISOString()} ===\n\n`);
        res.writeHead(200);
        res.end('cleared');
        return;
    }

    res.writeHead(404);
    res.end('not found');
});

server.listen(PORT, () => {
    console.log(`Log server running on http://localhost:${PORT}`);
    console.log(`Logs will be written to: ${LOG_FILE}`);
    console.log('Press Ctrl+C to stop');
});

File: membrane-field-core/test/log_server.py
#!/usr/bin/env python3
"""Simple log server - Run with: python log_server.py
Then open testBundle.html in browser. Logs written to debug.log
"""

import http.server
import json
from datetime import datetime
from pathlib import Path

LOG_FILE = Path(__file__).parent / "debug.log"
PORT = 9999

class LogHandler(http.server.BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header("Access-Control-Allow-Origin", "*")
        self.send_header("Access-Control-Allow-Methods", "POST, OPTIONS")
        self.send_header("Access-Control-Allow-Headers", "Content-Type")
        self.end_headers()

    def do_POST(self):
        if self.path == "/clear":
            LOG_FILE.write_text(f"=== Log Cleared: {datetime.now().isoformat()} ===\n\n")
            self.send_response(200)
            self.send_header("Access-Control-Allow-Origin", "*")
            self.end_headers()
            self.wfile.write(b"cleared")
            return

        if self.path == "/log":
            content_length = int(self.headers.get("Content-Length", 0))
            body = self.rfile.read(content_length).decode("utf-8")
            try:
                data = json.loads(body)
                timestamp = datetime.now().strftime("%H:%M:%S.%f")[:-3]
                line = f"[{timestamp}] {data['msg']}\n"
                with open(LOG_FILE, "a") as f:
                    f.write(line)
                self.send_response(200)
                self.send_header("Access-Control-Allow-Origin", "*")
                self.end_headers()
                self.wfile.write(b"ok")
            except Exception as e:
                self.send_response(400)
                self.send_header("Access-Control-Allow-Origin", "*")
                self.end_headers()
                self.wfile.write(str(e).encode())
            return

        self.send_response(404)
        self.send_header("Access-Control-Allow-Origin", "*")
        self.end_headers()

    def log_message(self, format, *args):
        pass  # Suppress default logging

if __name__ == "__main__":
    LOG_FILE.write_text(f"=== Debug Log Started: {datetime.now().isoformat()} ===\n\n")
    print(f"Log server running on http://localhost:{PORT}")
    print(f"Logs will be written to: {LOG_FILE}")
    print("Press Ctrl+C to stop")

    server = http.server.HTTPServer(("", PORT), LogHandler)
    server.serve_forever()
